### PR TITLE
docs: Update contrib guide for current branching

### PR DESCRIPTION
**Pull request by VladimirSlavik**
Followup to #4270

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -51,25 +51,18 @@ Which is my target git branch?
...
-need to open a separate pull-request for ``master`` in that case.
 
-If a change only applies to a Fedora XX release and the changes don't have to be merged back into ``master``,
-then the target branch should be a ``fXX-release`` relese branch.
+Notice that if the change applies to both Fedora Rawhide and Fedora XX, then you must create two
+pull requests, one for each target.
 
-The ``rhel-x`` branches on the other hand are not merged back into ``master``, and so a separate pull request
-has to be opened targeting either a ``fXX-devel`` branch or ``master``.
+The ``rhel-X`` branches on the other hand are not merged back into ``master``, and so a separate pull request
```
Comment (poncovka) [inline CONTRIBUTING.rst:64]: This is the same principal as above, so `on the other hand` doesn't make a lot of sense to me here.

**Review by poncovka — APPROVED** [review]
Thank you!

**Comment by VladimirSlavik** [comment]
This needs more work :(

**Comment by VladimirSlavik** [comment]
Re-reviewing the whole thing, it seems clear that "more work" would mean restructuring the whole docs bundle. For describing the current branching, this is fine though.

**Pull request merged**